app/oauth.go

```diff
@@ -80,7 +82,7 @@ func (a *App) DeleteOAuthApp(appId string) *model.AppError {
 		return err
 	}
 
-	a.InvalidateAllCaches()
+	_ = a.InvalidateAllCaches()
```

Comment:
What's the purpose of the three changes like this? Is this considered a best practice?

Reply:
I find it much easier to find bugs if returns are explicitly ignored. I've done changes like this also in the other idiomatic error handling PR.

But I wouldn't consider it "universal best practice". Effective Go even demands to check every returned error. (https://golang.org/doc/effective_go.html#blank_assign)

Reply:
Exactly, so why not check the error rather than discard it?

Reply:
Because this would be a behaviour change out of scope of this PR. But if you want, I can do so.

Reply:
Fixed. Please take a look. `AppError` always confuses me. I never know if i need a new error message or not.